packages/svelte/src/compiler/warnings.js

```diff
@@ -191,12 +191,20 @@ const state = {
 		`State referenced in its own scope will never update. Did you mean to reference it inside a closure?`
 };
 
+/** @satisfies {Warnings} */
+const performance = {
+	'inline-new-class': () =>
+		`Creating inline classes will likely cause performance issues. ` +
+		`Instead, declare the class at the module-level and create new instances from the class reference.`
```

Comment:
Should we have something in here like "or use getters/setters instead" because many people use it as a shortcut for this?

Reply:
Do we want to tell people to use getters/setters here, or to just hoist their class?